src/extension/background-script/connectors/lnd.ts

```diff
@@ -223,11 +227,76 @@ class Lnd implements Connector {
     });
   };
 
-  getTransactions = () => {
-    return this.request("GET", "/v1/payments", undefined, {
-      transactions: [],
-    });
-  };
+  async getInvoices(): Promise<GetInvoicesResponse> {
+    const data = await this.request<{
+      invoices: {
+        add_index: string;
+        amt_paid_msat: string;
+        amt_paid_sat: string;
+        amt_paid: string;
+        cltv_expiry: string;
+        creation_date: string;
+        description_hash?: string;
+        expiry: string;
+        fallback_addr: string;
+        features: unknown[];
+        htlcs: {
+          chan_id: string;
+          htlc_index: string;
+          amt_msat: string;
+          accept_height: number;
+          accept_time: string;
+          resolve_time: string;
+          expiry_height: number;
+          state: "SETTLED";
+          custom_records: Invoice["custom_records"];
+          mpp_total_amt_msat: string;
+          amp?: unknown;
+        }[];
+        is_keysend: boolean;
+        memo: string;
+        payment_addr: string;
+        payment_request: string;
+        private: boolean;
+        r_hash: string;
+        r_preimage: string;
+        route_hints: [];
+        settle_date: string;
+        settle_index: string;
+        settled: boolean;
+        state: string;
+        value_msat: string;
+        value: string;
+      }[];
+      last_index_offset: string;
+      first_index_offset: string;
+    }>("GET", "/v1/invoices", undefined, {});
+
+    const invoices: Invoice[] = data.invoices
+      .map((invoice, index): Invoice => {
+        const custom_records =
+          invoice.htlcs[0] && invoice.htlcs[0].custom_records;
+
+        return {
+          custom_records,
+          id: `${invoice.payment_request}-${index}`,
+          memo: invoice.memo,
+          preimage: invoice.r_preimage,
+          settled: invoice.settled,
+          settleDate: parseInt(invoice.settle_date) * 1000,
+          totalAmount: invoice.value,
```

Comment:
is LND returning strings here?

Reply:
Yes :(

https://api.lightning.community/?javascript#v1-invoices
![image](https://user-images.githubusercontent.com/1016218/178908651-de43f875-03f1-46c4-9d63-92b8c10a987f.png)